File: 2023-11-23-revisao-de-python/07-le-sequencia-de-inteiros-e-mostra-media.c
// '''
//     Função principal: 
//     - lê números inteiros até o usuário digitar um valor negativo; 
//     - armazena os valores lidos (exceto o valor negativo) em uma lista
//     - em seguida, exibe todos os valores lidos, bem como a média aritmética desses valores.
// '''
#include <stdlib.h>
#include <stdio.h>
#include <stdbool.h>

int mediaAritmeticaDeUmaLista(int* listaDeInteiros, int tamanhoDaLista) {
    int soma = 0;

    for (int i = 0; i < tamanhoDaLista; i++)
        soma += listaDeInteiros[i];

    return soma / tamanhoDaLista;
}

int main() {
    printf("Digite uns números inteiros aí. Qando terminar, digite um número negativo\n");
    int* listaDeInteiros = NULL;

    int quantidadeDeInteiros = 0;
    int inteiroDigitado = 0;

    while(true) {
        scanf("%d", &inteiroDigitado);
        if (inteiroDigitado <= 0) break;

        listaDeInteiros = realloc(listaDeInteiros, (quantidadeDeInteiros + 1) * sizeof(int));
        listaDeInteiros[quantidadeDeInteiros] = inteiroDigitado;
        quantidadeDeInteiros++;
    }

    int media = mediaAritmeticaDeUmaLista(listaDeInteiros, quantidadeDeInteiros);
    printf("\nA média artimética dos valores digitados é: %d", media);
}

File: 2023-11-23-revisao-de-python/07-le-sequencia-de-inteiros-e-mostra-media.py
'''
    Função principal: 
    - lê números inteiros até o usuário digitar um valor negativo; 
    - armazena os valores lidos (exceto o valor negativo) em uma lista
    - em seguida, exibe todos os valores lidos, bem como a média aritmética desses valores.
'''

def mediaAritmeticaDeUmaLista(listaDeInteiros: list) -> int:
    soma = 0
    for inteiro in listaDeInteiros:
        soma += inteiro

    return soma // len(listaDeInteiros)

def main() -> None:
    print("Digite uns números inteiros aí. Qando terminar, digite um número negativo")
    listaDeInteiros: list = []

    inteiro: int = 0
    while(True):
        inteiro = int(input())
        if inteiro <= 0:
            break

        listaDeInteiros.append(inteiro)

    print(f"\nOs números digitados foram: {listaDeInteiros}")
    print(f"\nA média artimética dos valores digitados é: {mediaAritmeticaDeUmaLista(listaDeInteiros)}")

if __name__ == "__main__":
    main()